packages/google-pay-integration/src/google-pay-button-strategy.ts

```diff
@@ -0,0 +1,257 @@
+} from './types';
+
+export default class GooglePayButtonStrategy implements CheckoutButtonStrategy {
+    private _methodId?: keyof WithGooglePayPaymentInitializeOptions;
+    private _buyNowCart?: Cart;
+    private _buyNowInitializeOptions?: GooglePayBuyNowInitializeOptions;
+    private _currencyCode?: string;
+
+    constructor(
+        private _paymentIntegrationService: PaymentIntegrationService,
+        private _googlePayPaymentProcessor: GooglePayPaymentProcessor,
+    ) {}
+
+    async initialize(
+        options: CheckoutButtonInitializeOptions & WithGooglePayButtonInitializeOptions,
+    ): Promise<void> {
+        if (!options.methodId || !isGooglePayKey(options.methodId)) {
+            throw new InvalidArgumentError(
+                'Unable to proceed because "methodId" is not a valid key.',
+            );
+        }
+
+        this._methodId = options.methodId;
+
+        if (!options.containerId) {
+            throw new InvalidArgumentError(
+                'Unable to proceed because "containerId" is not a valid key.',
+            );
+        }
+
+        const googlePayOptions = options[this._getMethodOrThrow()];
+
+        if (!googlePayOptions) {
+            throw new InvalidArgumentError('Unable to proceed without valid options.');
+        }
+
+        const { buyNowInitializeOptions, currencyCode } = googlePayOptions;
+
+        try {
+            if (buyNowInitializeOptions) {
+                if (!currencyCode) {
+                    throw new InvalidArgumentError(
+                        `Unable to initialize payment because "options.currencyCode" argument is not provided.`,
+                    );
+                }
+
+                this._currencyCode = currencyCode;
+                this._buyNowInitializeOptions = buyNowInitializeOptions;
+
+                await this._googlePayPaymentProcessor.initialize(
+                    () =>
+                        this._paymentIntegrationService
+                            .getState()
+                            .getPaymentMethodOrThrow<GooglePayInitializationData>(
+                                this._getMethodOrThrow(),
+                            ),
+                    this._getGooglePayClientOptions(googlePayOptions),
+                    !!buyNowInitializeOptions,
+                    currencyCode,
+                );
+            } else {
+                await this._paymentIntegrationService.loadDefaultCheckout();
+                await this._googlePayPaymentProcessor.initialize(() =>
+                    this._paymentIntegrationService
+                        .getState()
+                        .getPaymentMethodOrThrow<GooglePayInitializationData>(
+                            this._getMethodOrThrow(),
+                        ),
+                );
+            }
+        } catch (error) {
+            return;
+        }
+
+        this._addPaymentButton({
+            ...googlePayOptions,
+            container: options.containerId,
+        });
```

Comment:
Should the button be rendered on the page if there will be an error after loading payment method?

Reply:
So it might be better to place the method call inside `try` for better understanding for other devs